hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/BasicValuedCollectionPart.java

```diff
@@ -185,10 +186,13 @@ public Fetch generateFetch(
 				collectionDescriptor.getRole(),
 				nature.getName()
 		);
-
+		NavigablePath parent = fetchablePath.getParent();
+		if ( parent.getFullPath().endsWith( EntityIdentifierMapping.ROLE_LOCAL_NAME ) ) {
```

Comment:
Is there such possibility that if the path happens to end with `EntityIdentifierMapping.ROLE_LOCAL_NAME`, but not with `.EntityIdentifierMapping.ROLE_LOCAL_NAME`? Maybe there is more structural way to guard against such possibility which is based on `equals`, not `endsWith`?
Ignore my 2 cents comment if irrelevant, though.